src/nvim/os/shell.c

```diff
@@ -236,27 +235,27 @@ static int do_os_system(char **argv,
   // streams while there's still data in the OS buffer (due to the process
   // exiting before all data is read).
   if (input != NULL) {
-    proc->in->events = NULL;
-    wstream_init(proc->in, 0);
+    proc->in.events = NULL;
+    wstream_init(&proc->in, 0);
   }
-  proc->out->events = NULL;
-  rstream_init(proc->out, 0);
-  rstream_start(proc->out, data_cb, &buf);
-  proc->err->events = NULL;
-  rstream_init(proc->err, 0);
-  rstream_start(proc->err, data_cb, &buf);
+  proc->out.events = NULL;
+  rstream_init(&proc->out, 0);
+  rstream_start(&proc->out, data_cb, &buf);
+  proc->err.events = NULL;
```

Comment:
You right, strange it was written this in way to start with. (it made even less before when the streams weren't event owned by the proc)